great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -933,13 +933,7 @@ def execution_engine_type(self) -> Type[SqlAlchemyExecutionEngine]:
     def get_engine(self) -> sqlalchemy.Engine:
         if self.connection_string != self._cached_connection_string or not self._engine:
             try:
-                model_dict = self.dict(
-                    exclude=self._get_exec_engine_excludes(),
-                    config_provider=self._config_provider,
-                )
-                connection_string = model_dict.pop("connection_string")
-                kwargs = model_dict.pop("kwargs", {})
-                self._engine = sa.create_engine(connection_string, **kwargs)
+                self._create_engine()
```

Comment:
```suggestion
                self._engine = self._create_engine()
```

Thanks for breaking out this method. Much easier to reason about and extend.

Can you make this one small chance so we aren't modifying state and returning an `sqlalchemy.Engine`?